tools/accuracy_checker/accuracy_checker/metrics/metric.py

```diff
@@ -129,7 +130,15 @@ def _update_state(self, fn, state_key, default_factory=None):
 
     def _resolve_representation_containers(self, annotation, prediction):
         def get_resolve_subject(representation, source=None):
-            if not isinstance(representation, ContainerRepresentation):
+            def is_container(representation):
+                if isinstance(representation, ContainerRepresentation):
+                    return True
+                representation_parents = type(representation).__bases__
```

Comment:
I would suggest using `__mro__` instead of `__bases__`, because:

1. It includes indirect base classes, which better mirrors the original `isinstance` check.
2. It includes the class itself, which will make the preceding `if` unnecessary.

Reply:
Completely agree, thanks

Reply:
You didn't remove the redundant `if`, though.

Reply:
fixed